pipelines/api/utils.py

```diff
@@ -77,12 +79,25 @@ def _slugify_file(filename):
     basename = filename.rsplit('/', 1)[-1]
     return basename.rsplit('.', 1)[0]
 
-
 def _run_id_iterator(slug):
     for sub_folder in os.listdir(slug):
         if _is_valid_uuid(sub_folder):
             yield sub_folder
 
+# patched with limit, order by modify time
+def _run_id_iterator_with_limit(slug, limit):
+    prefix = slug.rstrip('/')
+    tgt = glob.glob('{}/*'.format(prefix))
+    # filter valid uuid
+    it = filter(lambda x: _is_valid_uuid(os.path.basename(x)), tgt)
+    # order by modify time, newer first
+    it = sorted(it, key=lambda x: os.path.getmtime(x) * -1)
+    # limit
+    tgt = list(it)[:limit]
```

Comment:
if the limit == 0 then it will return empty list, but usually ppl want to return all if they set a  limit parameter = 0